src/wmtk/Mesh.hpp

```diff
@@ -24,6 +24,31 @@ class Mesh
     friend class ParaviewWriter;
     friend class MeshReader;
 
+
+    // Interface for SyncMesh
+
+    // TODO: Need a wrapper for MeshAttributeHandle<Tuple>?
+
+    // size of dim(Mesh), store base_tuple for each simplex
+    std::vector<MeshAttributeHandle<long>> base_tuple_handle;
+
+    // if parent_mesh is not null, then this mesh is a child_mesh of parent_mesh
+    std::shared_ptr<Mesh> parent_mesh;
```

Comment:
Should just be a raw ptr or maybe a weak ptr

Reply:
I don't like the pointer to parent

Reply:
Since the submeshes do not need to reference the parent (it is the parent calling functions in them), what about moving all functions related to submeshes in the parent mesh, and replacing this pointer with a bool indicating if it is a submesh or not? Or maybe even the bool is not necessary.